File: services/education/internal/model/classroom.go
package model

import "time"

type ClassroomQueryDTO struct {
	EnabledOnly *bool  `json:"enabledOnly,omitempty"`
	SearchKey   string `json:"searchKey,omitempty"`
}

type ClassroomVO struct {
	ID         int64      `json:"id"`
	UUID       string     `json:"uuid,omitempty"`
	Version    int64      `json:"version,omitempty"`
	InstID     int64      `json:"instId"`
	Name       string     `json:"name"`
	Address    string     `json:"address,omitempty"`
	Enabled    bool       `json:"enabled"`
	CreateTime *time.Time `json:"createTime,omitempty"`
	UpdateTime *time.Time `json:"updateTime,omitempty"`
}

type ClassroomMutation struct {
	ID      *int64 `json:"id,omitempty"`
	UUID    string `json:"uuid,omitempty"`
	Version *int64 `json:"version,omitempty"`
	Name    string `json:"name"`
	Address string `json:"address,omitempty"`
	Enabled *bool  `json:"enabled,omitempty"`
}

type ClassroomStatusMutation struct {
	ID      *int64 `json:"id,omitempty"`
	Enabled *bool  `json:"enabled,omitempty"`
}

type ClassroomDeleteDTO struct {
	ID *int64 `json:"id,omitempty"`
}

File: 5.0-admin/src/api/business-settings/classroom.ts
import { useGet, usePost } from '~/utils/request'

export interface ClassroomItem {
  id: number
  uuid?: string
  version?: number
  instId?: number
  name: string
  address?: string
  capacity?: number
  enabled: boolean
  remark?: string
  sort?: number
  createTime?: string
  updateTime?: string
}

export interface ClassroomMutation {
  id?: number
  uuid?: string
  version?: number
  name: string
  address?: string
  capacity: number
  enabled?: boolean
  remark?: string
  sort?: number
}

export function listClassroomsApi(params?: {
  enabledOnly?: boolean
  searchKey?: string
}) {
  return useGet<ClassroomItem[]>('/api/v1/classrooms', params)
}

export function createClassroomApi(data: ClassroomMutation) {
  return usePost<{ id: number }>('/api/v1/classrooms/create', data)
}

export function updateClassroomApi(data: ClassroomMutation) {
  return usePost<boolean>('/api/v1/classrooms/update', data)
}

export function updateClassroomStatusApi(data: {
  id: number
  enabled: boolean
}) {
  return usePost<boolean>('/api/v1/classrooms/status', data)
}

export function deleteClassroomApi(data: {
  id: number
}) {
  return usePost<boolean>('/api/v1/classrooms/delete', data)
}
